Crash when illegal file encoding `utf` in the first line

### Bug description

When parsing the following file:

<!--
 If sharing the code is not an option, please state so,
 but providing only the stacktrace would still be helpful.
 -->

```python
# -*- encoding: utf -*-

```

pylint crashed with a ``AstroidError`` and with the following stacktrace:
```
Traceback (most recent call last):
  File "/home/ubuntu/.local/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 782, in _lint_file
    check_astroid_module(module)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 1052, in check_astroid_module
    retval = self._check_astroid_module(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 1098, in _check_astroid_module
    raw_checker.process_module(node)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/pylint/checkers/unicode.py", line 531, in process_module
    line = _remove_bom(line, codec)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/pylint/checkers/unicode.py", line 223, in _remove_bom
    bom = UNICODE_BOMS[encoding]
KeyError: 'utf'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ubuntu/.local/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 747, in _lint_files
    self._lint_file(fileitem, module, check_astroid_module)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 784, in _lint_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError
```


### Configuration

_No response_

### Command used

```shell
pylint main.py
```


### Pylint output

```shell
************* Module main
main.py:1:0: C2503: PEP8 recommends UTF-8 as encoding for Python files (bad-file-encoding)
main.py:1:0: F0002: main.py: Fatal error while checking 'main.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '/home/ubuntu/.cache/pylint/pylint-crash-2022-10-22-01-14-12.txt'. (astroid-error)
```


### Expected behavior

pylint should show the error about the illegal file encoding `utf` and do not crash

### Pylint version

```shell
pylint 2.15.4
astroid 2.12.12
Python 3.8.10 (default, Mar 15 2022, 12:22:08) 
[GCC 9.4.0]
```


### OS / Environment

Linux 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
Ubuntu 20.04 LTS

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/7661.bugfix
+++ b/doc/whatsnew/fragments/7661.bugfix
@@ -1,3 +1,3 @@
-Fix bug that unsafely retrieved a value from a dict and led to an ``astroid-error`` when parsing a module.
+Fix crash that happened when parsing files with unexpected encoding starting with 'utf' like ``utf13``
 
 Closes #7661

--- a/doc/whatsnew/fragments/7661.bugfix
+++ b/doc/whatsnew/fragments/7661.bugfix
@@ -1,3 +1,3 @@
-Fix crash that happened when parsing files with unexpected encoding starting with 'utf' like ``utf13``
+Fix crash that happened when parsing files with unexpected encoding starting with 'utf' like ``utf13``.
 
 Closes #7661

--- a/doc/whatsnew/fragments/7661.bugfix
+++ b/doc/whatsnew/fragments/7661.bugfix
@@ -1,0 +1,3 @@
+Fix bug that unsafely retrieved a value from a dict and led to an ``astroid-error`` when parsing a module.
+
+Closes #7661

--- a/pylint/checkers/unicode.py
+++ b/pylint/checkers/unicode.py
@@ -218,7 +218,7 @@ def _normalize_codec_name(codec: str) -> str:
 
 def _remove_bom(encoded: bytes, encoding: str) -> bytes:
     """Remove the bom if given from a line."""
-    if not encoding.startswith("utf"):
+    if encoding not in UNICODE_BOMS:
         return encoded
     bom = UNICODE_BOMS[encoding]
     if encoded.startswith(bom):

--- a/tests/regrtest_data/encoding/bad_missing_num.py
+++ b/tests/regrtest_data/encoding/bad_missing_num.py
@@ -1,0 +1,1 @@
+# -*- encoding: utf -*-

--- a/tests/regrtest_data/encoding/bad_wrong_num.py
+++ b/tests/regrtest_data/encoding/bad_wrong_num.py
@@ -1,0 +1,1 @@
+# -*- encoding: utf-9 -*-

--- a/tests/regrtest_data/encoding/good.py
+++ b/tests/regrtest_data/encoding/good.py
@@ -1,0 +1,1 @@
+# -*- encoding: utf-8 -*-

--- a/tests/test_self.py
+++ b/tests/test_self.py
@@ -148,13 +148,18 @@ class TestRunTC:
         output = re.sub(CLEAN_PATH, "", output, flags=re.MULTILINE)
         return output.replace("\\", "/")
 
-    def _test_output(self, args: list[str], expected_output: str) -> None:
+    def _test_output(
+        self, args: list[str], expected_output: str, unexpected_output: str = ""
+    ) -> None:
         out = StringIO()
         args = _add_rcfile_default_pylintrc(args)
         self._run_pylint(args, out=out)
         actual_output = self._clean_paths(out.getvalue())
         expected_output = self._clean_paths(expected_output)
         assert expected_output.strip() in actual_output.strip()
+
+        if unexpected_output:
+            assert unexpected_output.strip() not in actual_output.strip()
 
     def _test_output_file(
         self, args: list[str], filename: LocalPath, expected_output: str
@@ -1195,6 +1200,20 @@ a.py:1:4: E0001: Parsing failed: 'invalid syntax (<unknown>, line 1)' (syntax-er
         module = join(HERE, "regrtest_data", "invalid_encoding.py")
         expected_output = "unknown encoding"
         self._test_output([module, "-E"], expected_output=expected_output)
+
+    @pytest.mark.parametrize(
+        "module_name,expected_output",
+        [
+            ("good.py", ""),
+            ("bad_wrong_num.py", "(syntax-error)"),
+            ("bad_missing_num.py", "(bad-file-encoding)"),
+        ],
+    )
+    def test_encoding(self, module_name: str, expected_output: str) -> None:
+        path = join(HERE, "regrtest_data", "encoding", module_name)
+        self._test_output(
+            [path], expected_output=expected_output, unexpected_output="(astroid-error)"
+        )
 
 
 class TestCallbackOptions:

